modules/couchbase/src/main/java/org/testcontainers/couchbase/CouchbaseContainer.java

```diff
@@ -15,447 +15,423 @@
+        for (BucketDefinition bucket : buckets) {
+            logger().debug("Creating bucket \"" + bucket.getName() + "\"");
 
-    public CouchbaseContainer withMemoryQuota(String memoryQuota) {
-        this.memoryQuota = memoryQuota;
-        return self();
-    }
-
-    public CouchbaseContainer withIndexMemoryQuota(String indexMemoryQuota) {
-        this.indexMemoryQuota = indexMemoryQuota;
-        return self();
-    }
+            Response response = doHttpRequest(MGMT_PORT, "/pools/default/buckets", "POST", new FormBody.Builder()
+                .add("name", bucket.getName())
+                .add("ramQuotaMB", Integer.toString(bucket.getQuota()))
+                .build(), true);
 
-    public CouchbaseContainer withClusterAdmin(String username, String password) {
-        this.clusterUsername = username;
-        this.clusterPassword = password;
-        return self();
-    }
+            checkSuccessfulResponse(response, "Could not create bucket " + bucket.getName());
 
-    public CouchbaseContainer withKeyValue(boolean keyValue) {
-        this.keyValue = keyValue;
-        return self();
-    }
+            new HttpWaitStrategy()
+                .forPath("/pools/default/buckets/" + bucket.getName())
+                .forPort(MGMT_PORT)
+                .withBasicCredentials(username, password)
+                .forStatusCode(200)
+                .waitUntilReady(this);
 
-    public CouchbaseContainer withQuery(boolean query) {
-        this.query = query;
-        return self();
-    }
+            if (bucket.hasPrimaryIndex()) {
+                if (enabledServices.contains(CouchbaseService.QUERY)) {
+                    Response queryResponse = doHttpRequest(QUERY_PORT, "/query/service", "POST", new FormBody.Builder()
+                        .add("statement", "CREATE PRIMARY INDEX on `" + bucket.getName() + "`")
+                        .build(), true);
 
-    public CouchbaseContainer withIndex(boolean index) {
-        this.index = index;
-        return self();
+                    checkSuccessfulResponse(queryResponse, "Could not create primary index for bucket " + bucket.getName());
+                } else {
+                    logger().info("Primary index creation for bucket " + bucket.getName() + " ignored, since QUERY service is not present.");
+                }
+            }
+        }
     }
 
-    public CouchbaseContainer withPrimaryIndex(boolean primaryIndex) {
-        this.primaryIndex = primaryIndex;
-        return self();
+    /**
+     * Helper method to extract the internal IP address based on the network configuration.
+     */
+    private String getInternalIpAddress() {
+        final Map<String, ContainerNetwork> networks = getContainerInfo().getNetworkSettings().getNetworks();
+        for (ContainerNetwork network : networks.values()) {
+            return network.getIpAddress();
+        }
+        throw new IllegalStateException("No network available to extract the internal IP from!");
```

Comment:
```suggestion
        return getContainerInfo().getNetworkSettings().getNetworks().values().stream()
            .map(ContainerNetwork::getIpAddress)
            .findFirst()
            .orElseThrow(() -> new IllegalStateException("No network available to extract the internal IP from!"));
```

Not sure if better, but on first reading, I stumbled upon the for-loop for getting the first element.